crates/topos-tce-api/src/stream/mod.rs

```diff
@@ -31,6 +31,38 @@ pub use self::commands::StreamCommand;
 pub use self::errors::StreamError;
 pub(crate) use self::errors::{HandshakeError, StreamErrorKind};
 
+/// [`TransientStream`] is a stream that live as long as the connection is open.
+/// A [`TransientStream`] will not receive any certificates that were delivered
+/// before the stream was ready to listen.
+///
+/// [`TransientStream`] is implementing [`futures::Stream`] and use a custom [`Drop`]
+/// implementation to notify the `runtime` when ended.
+#[derive(Debug)]
+pub struct TransientStream {
+    pub(crate) inner: mpsc::Receiver<Certificate>,
+    pub(crate) stream_id: Uuid,
+    pub(crate) notifier: Option<oneshot::Sender<Uuid>>,
```

Comment:
What does having a `None` here mean? Can't we guarantee that we always create these with a notifier?

Reply:
It's a `oneshot::Sender` it moves when used, wrapping it in `Option` allows you to consume it one time exactly.